freight/providers/pipeline.py

```diff
@@ -674,7 +679,11 @@ def rollout_status_deployment(
     for condition in deployment.status.conditions or []:
         if condition.type == "Progressing":
             if condition.reason == "ProgressDeadlineExceeded":
-                return f"deployment {repr(name)} exceeded its progress deadline", False
+                # Experimental deploy will return True as success response for the deploy regardless of failure
+                return (
+                    f"deployment {repr(name)} exceeded its progress deadline",
```

Comment:
This makes sense, but I wonder if there will be any way to know if an experimental step failed without scanning the freight logs.